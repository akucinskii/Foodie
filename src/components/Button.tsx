import React from "react";

type Props = {
  onClick?: () => void;
  color?: string;
  children: React.ReactNode;
};

const Button = ({ onClick, children }: Props) => {
  return (
    <button
      type="button"
      className={`sm:text:sm md:text-md inline-block rounded bg-yellow-500 px-6 py-2.5 text-xs font-medium uppercase leading-tight text-white shadow-md transition duration-150 ease-in-out hover:bg-yellow-600 hover:shadow-lg focus:bg-yellow-600 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-yellow-700 active:shadow-lg`}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default Button;
