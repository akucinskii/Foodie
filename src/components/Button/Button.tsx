import React from "react";

type Props = {
  onClick?: () => void;
  children: React.ReactNode;
  disabled?: boolean;
  className?: string;
};

const Button = ({ onClick, children, disabled, className }: Props) => {
  return (
    <button
      type="button"
      data-testid="button"
      className={className ? className : `btn btn-primary text-white`}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

export default Button;
