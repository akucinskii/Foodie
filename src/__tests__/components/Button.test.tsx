import { render } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";
import Button from "../../components/Button";

describe("Button", () => {
  const mockProp = {
    onClick: vi.fn(),
    children: "Test",
    disabled: false,
  };
  const { container, rerender, getByTestId } = render(<Button {...mockProp} />);

  test("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  test("should call onClick when clicked", () => {
    rerender(<Button {...mockProp} />);
    getByTestId("button").click();
    expect(mockProp.onClick).toHaveBeenCalled();
  });

  test("should be disabled when disabled is true", () => {
    rerender(<Button {...mockProp} disabled={true} />);
    expect(getByTestId("button").getAttribute("disabled")).toBeDefined();
  });

  test("should not be disabled when disabled is false", () => {
    rerender(<Button {...mockProp} disabled={false} />);
    expect(getByTestId("button").getAttribute("disabled")).toBeNull();
  });

  test("should have children", () => {
    rerender(<Button {...mockProp} />);
    expect(getByTestId("button").textContent).toBe("Test");
  });
});
