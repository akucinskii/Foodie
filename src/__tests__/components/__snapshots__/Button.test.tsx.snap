// Vitest Snapshot v1

exports[`Button > should match snapshot 1`] = `
<div>
  <button
    class="btn btn-primary text-white"
    data-testid="button"
    type="button"
  >
    Test
  </button>
</div>
`;
